### PR TITLE
Insight Redesign

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- Insights redesign implementation

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

#### Screenshots (optional)
Screenshot 2023-03-16 at 15 53 40

### CODE REVIEW

**Comment by Baalmart** [comment]
Thanks for resolving the merge conflicts in a timely manner! @NoahNsimbe 

- Please add an appropriate PR description, appears the current is just the unedited template notes. :)
- You can also add just a few PR review comments to guide the reviewers around the key changes in the codebase.

File: mobile/lib/constants/api.dart [hunk 1]
```diff
@@ -6,6 +6,8 @@ class AirQoUrls {
   static String get insights =>
       '${Config.airqoApiUrl}view/measurements/mobile-app/insights';
 
+  static String get forecast => '${Config.airqoApiUrl}predict/daily/';
```
Comment (NoahNsimbe) [inline mobile/lib/constants/api.dart:9]: @Mnoble-19 endpoint expected to provide daily forecast

File: mobile/lib/models/database.dart [hunk 2]
```diff
@@ -42,92 +31,52 @@ class AirQoDatabase extends _$AirQoDatabase {
...
+      },
+      onUpgrade: (Migrator m, int from, int to) async {
+        if (from < 2) {
+          await m.deleteTable('forecast_insights');
+          await m.deleteTable('historical_insights');
+        }
+
+        await m.createAll();
+      },
+    );
```
Comment (NoahNsimbe) [inline mobile/lib/models/database.dart:50]: Database migrations

File: mobile/lib/services/rest_api.dart [hunk 3]
```diff
@@ -165,36 +165,34 @@ class AirqoApiClient {
...
-      )}T23:59:59Z';
-
-      final queryParams = <String, dynamic>{
-        'siteId': siteId,
-        'utcOffset': now.getUtcOffset(),
-        'startDateTime': startDateTime,
-        'endDateTime': endDateTime,
-      };
+      final body =
+          await _performGetRequest({}, "${AirQoUrls.forecast}/$siteId");
```
Comment (NoahNsimbe) [inline mobile/lib/services/rest_api.dart:171]: @Mnoble-19 expecting the to be developed daily forecast endpoint to support querying by siteId.

File: mobile/lib/services/rest_api.dart [hunk 4]
```diff
@@ -165,36 +165,34 @@ class AirqoApiClient {
...
+      final forecast = <Forecast>[];
+
+      for (final prediction in body['predictions']) {
+        try {
+          forecast.add(
+            Forecast.fromJson({
+              'pm2_5': prediction['prediction_time'],
+              'time': prediction['prediction_value'],
+              'siteId': siteId,
+            }),
```
Comment (NoahNsimbe) [inline mobile/lib/services/rest_api.dart:182]: @Mnoble-19 expecting the daily forecast endpoint to return to return data with at least `prediction_time` and `prediction_value`

**Pull request merged**